From 16af7936ee7836d37c39457250479845a1c1f842 Mon Sep 17 00:00:00 2001
From: Kemeng Shi <shikemeng@huawei.com>
Date: Sat, 8 Jan 2022 18:52:28 +0800
Subject: [PATCH] etmem scan: fix memleak in vm_idle_read
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 16af7936ee7836d37c39457250479845a1c1f842


euleros inclusion
category: feature
feature: etmem
bugzilla: https://gitee.com/openeuler/kernel/issues/I4OODH?from=project-issue
CVE: NA

-------------------------------------------------

free pic before return from vm_idle_read in etmem scan

Signed-off-by: Kemeng Shi <shikemeng@huawei.com>
Reviewed-by: louhongxiang <louhongxiang@huawei.com>
Reviewed-by: Chen Wandun <chenwandun@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/proc/etmem_scan.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/proc/etmem_scan.c b/fs/proc/etmem_scan.c
index 9d615381e9fa..6253d4e5a556 100644
--- a/fs/proc/etmem_scan.c
+++ b/fs/proc/etmem_scan.c
@@ -891,6 +891,7 @@ static ssize_t vm_idle_read(struct file *file, char *buf,
 	ret = pic->bytes_copied;
 	*ppos = pic->next_hva;
 out_kvm:
+	kfree(pic);
 	return ret;
 
 }
-- 
2.26.2

